src/legacy/core_plugins/kbn_vislib_vis_types/public/components/options/metrics_axes/category_axis_panel.tsx

```diff
@@ -0,0 +1,122 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { useCallback } from 'react';
+import { EuiPanel, EuiTitle, EuiSpacer, EuiAccordion } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+
+import { VisOptionsProps } from 'ui/vis/editors/default';
+import { BasicVislibParams, Axis } from '../../../types';
+import { SelectOption, SwitchOption } from '../../common';
+import { LabelOptions } from './label_options';
+import { Positions } from '../../../utils/collections';
+
+interface CategoryAxisPanelProps extends VisOptionsProps<BasicVislibParams> {
+  onPositionChanged: (position: Positions) => void;
+}
+
+function CategoryAxisPanel(props: CategoryAxisPanelProps) {
```

Comment:
Not 100% sure but there is only one category axis allowed, right? Could we only pass this single object down into the component to avoid all these `[0]` accessors? Then we only have to solve it once in the top level component

Reply:
On ui the first category axis is displayed only.
I applied the suggestion, please take a look at e4d45ee7775239d08cf06b1d7fd80379e1af5c4b.